guardrails/validator_service.py

```diff
@@ -93,14 +106,14 @@ def run_validators(self, validation_logs, validator_setup, value, metadata):
             result = validator_logs.validation_result
             if isinstance(result, FailResult):
                 value = self.perform_correction(
-                    [result], value, validator, validator_setup.on_fail
+                    [result], value, validator, validator.on_fail_descriptor
```

Comment:
nice catch, how exactly did you run it to find this bug? IIRC the only time SequentialValidatorService is used is if there's an async event loop present and guardrails is invoked synchronously

Reply:
we should probably (in the future) add a test for parity between the implementations of ValidatorService

Reply:
Ran it in a Jupyter notebook!